1_inventory/src/get_wqp_inventory.R

```diff
@@ -60,12 +60,19 @@ inventory_wqp <- function(grid, char_names, wqp_args = NULL, max_tries = 3){
   }) %>%
     bind_rows()
   
-  # Fetch missing CRS information from WQP
-  site_location_metadata <- dataRetrieval::whatWQPsites(
-    bBox = c(bbox$xmin, bbox$ymin, bbox$xmax, bbox$ymax)) %>%
-    select(MonitoringLocationIdentifier, HorizontalCoordinateReferenceSystemDatumName) %>%
-    filter(MonitoringLocationIdentifier %in% wqp_inventory$MonitoringLocationIdentifier)
-  
+  # Fetch missing CRS information from WQP. Note that an empty query will not
+  # contain CRS information. In the event that the lines below throw an error, 
+  # "Column `HorizontalCoordinateReferenceSystemDatumName` doesn't exist", return
+  # an empty data frame for the site location metadata. 
+  site_location_metadata <- tryCatch(
+    dataRetrieval::whatWQPsites(bBox = c(bbox$xmin, bbox$ymin, bbox$xmax, bbox$ymax)) %>%
+      select(MonitoringLocationIdentifier, HorizontalCoordinateReferenceSystemDatumName) %>%
+      filter(MonitoringLocationIdentifier %in% wqp_inventory$MonitoringLocationIdentifier),
+    error = function(e){
+      data.frame(MonitoringLocationIdentifier = character())
```

Comment:
Do we want to keep the datum column in the empty one? That way, the column still exists and future joining or row-binding won't fuss about it?
```suggestion
      data.frame(MonitoringLocationIdentifier = character(), HorizontalCoordinateReferenceSystemDatumName = character())
```

Reply:
Adding to this to say that this would only matter if _all_ of the branches within the single inventory pattern were empty (highly unlikely with how we have it set up). But if someone used this function and made their own separate targets rather than branching over everything, they may get some strange message if they try to combine on their own later (although, I am realizing that `tidyverse::bind_rows()` handles missing columns now compared to `rbind()`, so this might be moot.

Reply:
Yeah, I think I did leave the empty data frame minimal because of how `bind_rows()` handles missing columns, but I agree with your assessment and that we should explicitly add the datum column to avoid unexpected errors. Thanks for testing this out, @lindsayplatt!